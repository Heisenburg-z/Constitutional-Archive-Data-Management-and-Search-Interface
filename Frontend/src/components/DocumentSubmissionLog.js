import React from 'react';
import { History, User, Download, Trash2, Edit, FileText, Calendar } from 'lucide-react';

const DocumentSubmissionLog = ({
  documents = [],
  handleDownloadDocument = () => {},
  handleEditMetadata = () => {},
  setDocumentToDelete = () => {},
  downloadingDocs = {},
  isDeleting = false
}) => {
  const sortedDocuments = [...documents].sort((a, b) => 
    new Date(b.createdAt) - new Date(a.createdAt)
  );

  if (!Array.isArray(documents)) {
    return (
      <div className="bg-white rounded-lg shadow-md p-6">
        <div className="text-red-500">Error: Document data is not available</div>
      </div>
    );
  }

  return (
    <div className="bg-white rounded-lg shadow-md p-6 border border-gray-100">
      <div className="flex items-center gap-3 mb-6">
        <div className="p-2 rounded-lg bg-blue-50 text-blue-600">
          <History className="h-5 w-5" />
        </div>
        <div>
          <h2 className="text-xl font-bold text-gray-800">Document Submission Log</h2>
          <p className="text-sm text-gray-500">
            {documents.length === 0 
              ? "No documents submitted yet" 
              : "Track all document submissions with upload details"}
          </p>
        </div>
      </div>
      
      {sortedDocuments.length > 0 ? (
        <div className="overflow-x-auto">
          <table className="min-w-full divide-y divide-gray-200">
            <thead className="bg-gray-50">
              <tr>
                <th className="px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider">
                  <div className="flex items-center gap-1">
                    <FileText className="h-3.5 w-3.5" />
                    Document
                  </div>
                </th>
                <th className="px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider">
                  <div className="flex items-center gap-1">
                    <User className="h-3.5 w-3.5" />
                    Submitted By
                  </div>
                </th>
                <th className="px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider">
                  <div className="flex items-center gap-1">
                    <Calendar className="h-3.5 w-3.5" />
                    Submission Date
                  </div>
                </th>
                <th className="px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider">Type</th>
                <th className="px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider">Actions</th>
              </tr>
            </thead>
            <tbody className="bg-white divide-y divide-gray-200">
              {sortedDocuments.map((doc) => (
                <tr key={doc._id} className="hover:bg-gray-50 transition-colors">
                  <td className="px-6 py-4 whitespace-nowrap">
                    <div className="text-sm font-medium text-gray-900">
                      {doc.name}
                      {doc.metadata?.title && (
                        <div className="text-xs text-gray-500 mt-1">
                          {doc.metadata.title}
                        </div>
                      )}
                    </div>
                  </td>
                  <td className="px-6 py-4 whitespace-nowrap">
                    <div className="flex items-center">
                      <div className="text-sm">
                        <div className="font-medium text-gray-900">
                          {doc.createdBy?.firstName} {doc.createdBy?.lastName}
                        </div>
                        <div className="text-xs text-gray-500">
                          {doc.createdBy?.email}
                        </div>
                        {doc.createdBy?.role === 'admin' && (
                          <span className="mt-1 inline-flex items-center px-2 py-0.5 rounded text-xs font-medium bg-green-100 text-green-800">
                            Admin
                          </span>
                        )}
                      </div>
                    </div>
                  </td>
                  <td className="px-6 py-4 whitespace-nowrap">
                    <div className="text-sm text-gray-900">
                      {new Date(doc.createdAt).toLocaleDateString()}
                    </div>
                    <div className="text-xs text-gray-500">
                      {new Date(doc.createdAt).toLocaleTimeString()}
                    </div>
                  </td>
                  <td className="px-6 py-4 whitespace-nowrap">
                    <span className={`px-2 py-1 text-xs font-medium rounded-full ${
                      doc.type === 'file' ? 'bg-blue-100 text-blue-800' : 
                      doc.type === 'directory' ? 'bg-purple-100 text-purple-800' :
                      'bg-gray-100 text-gray-800'
                    }`}>
                      {doc.type}
                    </span>
                  </td>
                  <td className="px-6 py-4 whitespace-nowrap text-sm font-medium flex gap-2">
                    <button
                      onClick={() => handleDownloadDocument(doc)}
                      disabled={downloadingDocs[doc._id]}
                      className="text-blue-600 hover:text-blue-900 p-1.5 rounded hover:bg-blue-50 transition-colors"
                      title="Download"
                    >
                      <Download className="h-4 w-4" />
                    </button>
                    <button
                      onClick={() => handleEditMetadata(doc)}
                      className="text-green-600 hover:text-green-900 p-1.5 rounded hover:bg-green-50 transition-colors"
                      title="Edit"
                    >
                      <Edit className="h-4 w-4" />
                    </button>
                    <button
                      onClick={() => setDocumentToDelete(doc._id)}
                      disabled={isDeleting}
                      className="text-red-600 hover:text-red-900 p-1.5 rounded hover:bg-red-50 transition-colors"
                      title="Delete"
                    >
                      <Trash2 className="h-4 w-4" />
                    </button>
                  </td>
                </tr>
              ))}
            </tbody>
          </table>
        </div>
      ) : (
        <div className="text-center py-8 text-gray-500">
          <FileText className="mx-auto h-8 w-8 text-gray-300 mb-2" />
          <p>No documents have been submitted yet</p>
        </div>
      )}
    </div>
  );
};

export default DocumentSubmissionLog;