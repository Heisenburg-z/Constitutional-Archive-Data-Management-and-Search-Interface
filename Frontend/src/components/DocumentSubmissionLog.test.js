import React from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import DocumentSubmissionLog from './DocumentSubmissionLog';

// Mock lucide-react icons
jest.mock('lucide-react', () => ({
  History: () => <div data-testid="history-icon" />,
  User: () => <div data-testid="user-icon" />,
  Download: () => <div data-testid="download-icon" />,
  Trash2: () => <div data-testid="trash-icon" />,
  Edit: () => <div data-testid="edit-icon" />,
  FileText: () => <div data-testid="filetext-icon" />,
  Calendar: () => <div data-testid="calendar-icon" />
}));

describe('DocumentSubmissionLog Component', () => {
  const mockDocument = {
    _id: '1',
    name: 'test-document.pdf',
    type: 'file',
    createdAt: '2024-01-15T10:30:00.000Z',
    createdBy: {
      firstName: 'John',
      lastName: 'Doe',
      email: 'john.doe@example.com',
      role: 'user'
    },
    metadata: {
      title: 'Test Document Title'
    }
  };

  const mockAdminDocument = {
    _id: '2',
    name: 'admin-document.pdf',
    type: 'directory',
    createdAt: '2024-01-16T14:45:00.000Z',
    createdBy: {
      firstName: 'Admin',
      lastName: 'User',
      email: 'admin@example.com',
      role: 'admin'
    }
  };

  const defaultProps = {
    documents: [],
    handleDownloadDocument: jest.fn(),
    handleEditMetadata: jest.fn(),
    setDocumentToDelete: jest.fn(),
    downloadingDocs: {},
    isDeleting: false
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  describe('Component Rendering', () => {
    test('renders with default props', () => {
      render(<DocumentSubmissionLog {...defaultProps} />);
      
      expect(screen.getByText('Document Submission Log')).toBeInTheDocument();
      expect(screen.getByText('No documents submitted yet')).toBeInTheDocument();
    });

    test('renders header section correctly', () => {
      render(<DocumentSubmissionLog {...defaultProps} />);
      
      expect(screen.getByTestId('history-icon')).toBeInTheDocument();
      expect(screen.getByText('Document Submission Log')).toBeInTheDocument();
      expect(screen.getByText('No documents submitted yet')).toBeInTheDocument();
    });

    test('renders empty state when no documents', () => {
      render(<DocumentSubmissionLog {...defaultProps} />);
      
      expect(screen.getByText('No documents have been submitted yet')).toBeInTheDocument();
      expect(screen.getByTestId('filetext-icon')).toBeInTheDocument();
    });

    test('renders table when documents are provided', () => {
      const props = {
        ...defaultProps,
        documents: [mockDocument]
      };
      
      render(<DocumentSubmissionLog {...props} />);
      
      expect(screen.getByRole('table')).toBeInTheDocument();
      expect(screen.getByText('Document')).toBeInTheDocument();
      expect(screen.getByText('Submitted By')).toBeInTheDocument();
      expect(screen.getByText('Submission Date')).toBeInTheDocument();
      expect(screen.getByText('Type')).toBeInTheDocument();
      expect(screen.getByText('Actions')).toBeInTheDocument();
    });
  });

  describe('Error Handling', () => {
    test('handles non-array documents prop', () => {
      const props = {
        ...defaultProps,
        documents: null
      };
      
      render(<DocumentSubmissionLog/>);
      
      expect(screen.getByText('No documents submitted yet')).toBeInTheDocument();
    });

    test('handles undefined documents prop', () => {
      const props = {
        ...defaultProps,
        documents: undefined
      };
      
      render(<DocumentSubmissionLog {...props} />);
      
      expect(screen.getByText('No documents have been submitted yet')).toBeInTheDocument();
    });
  });

  describe('Document Display', () => {
    test('displays document information correctly', () => {
      const props = {
        ...defaultProps,
        documents: [mockDocument]
      };
      
      render(<DocumentSubmissionLog {...props} />);
      
      expect(screen.getByText('test-document.pdf')).toBeInTheDocument();
      expect(screen.getByText('Test Document Title')).toBeInTheDocument();
      expect(screen.getByText('John Doe')).toBeInTheDocument();
      expect(screen.getByText('john.doe@example.com')).toBeInTheDocument();
      expect(screen.getByText('file')).toBeInTheDocument();
    });

    test('displays admin badge for admin users', () => {
      const props = {
        ...defaultProps,
        documents: [mockAdminDocument]
      };
      
      render(<DocumentSubmissionLog {...props} />);
      
      expect(screen.getByText('Admin')).toBeInTheDocument();
      expect(screen.getByText('Admin')).toHaveClass('bg-green-100', 'text-green-800');
    });

    test('handles missing metadata gracefully', () => {
      const docWithoutMetadata = {
        ...mockDocument,
        metadata: undefined
      };
      
      const props = {
        ...defaultProps,
        documents: [docWithoutMetadata]
      };
      
      render(<DocumentSubmissionLog {...props} />);
      
      expect(screen.getByText('test-document.pdf')).toBeInTheDocument();
      expect(screen.queryByText('Test Document Title')).not.toBeInTheDocument();
    });

    test('handles missing createdBy information', () => {
      const docWithoutCreatedBy = {
        ...mockDocument,
        createdBy: {}
      };
      
      const props = {
        ...defaultProps,
        documents: [docWithoutCreatedBy]
      };
      
      render(<DocumentSubmissionLog {...props} />);
      
      expect(screen.getByText('test-document.pdf')).toBeInTheDocument();
    });
  });

  describe('Document Type Styling', () => {
    test('applies correct styling for file type', () => {
      const props = {
        ...defaultProps,
        documents: [mockDocument]
      };
      
      render(<DocumentSubmissionLog {...props} />);
      
      const typeElement = screen.getByText('file');
      expect(typeElement).toHaveClass('bg-blue-100', 'text-blue-800');
    });

    test('applies correct styling for directory type', () => {
      const props = {
        ...defaultProps,
        documents: [mockAdminDocument]
      };
      
      render(<DocumentSubmissionLog {...props} />);
      
      const typeElement = screen.getByText('directory');
      expect(typeElement).toHaveClass('bg-purple-100', 'text-purple-800');
    });

    test('applies default styling for unknown type', () => {
      const docWithUnknownType = {
        ...mockDocument,
        type: 'unknown'
      };
      
      const props = {
        ...defaultProps,
        documents: [docWithUnknownType]
      };
      
      render(<DocumentSubmissionLog {...props} />);
      
      const typeElement = screen.getByText('unknown');
      expect(typeElement).toHaveClass('bg-gray-100', 'text-gray-800');
    });
  });

  describe('Date Formatting', () => {
    test('formats dates correctly', () => {
      const props = {
        ...defaultProps,
        documents: [mockDocument]
      };
      
      render(<DocumentSubmissionLog {...props} />);
      
      // Check that date is displayed (format may vary by locale)
      const dateElements = screen.getAllByText(/2024|1\/15|15\/1/);
      expect(dateElements.length).toBeGreaterThan(0);
    });
  });

  describe('Document Sorting', () => {
    test('sorts documents by creation date (newest first)', () => {
      const olderDoc = {
        ...mockDocument,
        _id: 'older',
        name: 'older-document.pdf',
        createdAt: '2024-01-10T10:30:00.000Z'
      };
      
      const newerDoc = {
        ...mockDocument,
        _id: 'newer',
        name: 'newer-document.pdf',
        createdAt: '2024-01-20T10:30:00.000Z'
      };
      
      const props = {
        ...defaultProps,
        documents: [olderDoc, newerDoc]
      };
      
      render(<DocumentSubmissionLog {...props} />);
      
      const rows = screen.getAllByRole('row');
      // First row is header, second should be newer document
      expect(rows[1]).toHaveTextContent('newer-document.pdf');
      expect(rows[2]).toHaveTextContent('older-document.pdf');
    });
  });

  describe('Action Buttons', () => {
    test('renders all action buttons', () => {
      const props = {
        ...defaultProps,
        documents: [mockDocument]
      };
      
      render(<DocumentSubmissionLog {...props} />);
      
      expect(screen.getByTestId('download-icon')).toBeInTheDocument();
      expect(screen.getByTestId('edit-icon')).toBeInTheDocument();
      expect(screen.getByTestId('trash-icon')).toBeInTheDocument();
    });

    test('calls handleDownloadDocument when download button is clicked', () => {
      const mockHandleDownload = jest.fn();
      const props = {
        ...defaultProps,
        documents: [mockDocument],
        handleDownloadDocument: mockHandleDownload
      };
      
      render(<DocumentSubmissionLog {...props} />);
      
      const downloadButton = screen.getByTitle('Download');
      fireEvent.click(downloadButton);
      
      expect(mockHandleDownload).toHaveBeenCalledWith(mockDocument);
    });

    test('calls handleEditMetadata when edit button is clicked', () => {
      const mockHandleEdit = jest.fn();
      const props = {
        ...defaultProps,
        documents: [mockDocument],
        handleEditMetadata: mockHandleEdit
      };
      
      render(<DocumentSubmissionLog {...props} />);
      
      const editButton = screen.getByTitle('Edit');
      fireEvent.click(editButton);
      
      expect(mockHandleEdit).toHaveBeenCalledWith(mockDocument);
    });

    test('calls setDocumentToDelete when delete button is clicked', () => {
      const mockSetDocumentToDelete = jest.fn();
      const props = {
        ...defaultProps,
        documents: [mockDocument],
        setDocumentToDelete: mockSetDocumentToDelete
      };
      
      render(<DocumentSubmissionLog {...props} />);
      
      const deleteButton = screen.getByTitle('Delete');
      fireEvent.click(deleteButton);
      
      expect(mockSetDocumentToDelete).toHaveBeenCalledWith('1');
    });

    test('disables download button when document is being downloaded', () => {
      const props = {
        ...defaultProps,
        documents: [mockDocument],
        downloadingDocs: { '1': true }
      };
      
      render(<DocumentSubmissionLog {...props} />);
      
      const downloadButton = screen.getByTitle('Download');
      expect(downloadButton).toBeDisabled();
    });

    test('disables delete button when deletion is in progress', () => {
      const props = {
        ...defaultProps,
        documents: [mockDocument],
        isDeleting: true
      };
      
      render(<DocumentSubmissionLog {...props} />);
      
      const deleteButton = screen.getByTitle('Delete');
      expect(deleteButton).toBeDisabled();
    });
  });

  describe('Table Structure', () => {
    test('has correct table headers', () => {
      const props = {
        ...defaultProps,
        documents: [mockDocument]
      };
      
      render(<DocumentSubmissionLog {...props} />);
      
      expect(screen.getByTestId('filetext-icon')).toBeInTheDocument();
      expect(screen.getByTestId('user-icon')).toBeInTheDocument();
      expect(screen.getByTestId('calendar-icon')).toBeInTheDocument();
    });

    test('applies hover effects to table rows', () => {
      const props = {
        ...defaultProps,
        documents: [mockDocument]
      };
      
      render(<DocumentSubmissionLog {...props} />);
      
      const row = screen.getByRole('row', { name: /test-document.pdf/i });
      expect(row).toHaveClass('hover:bg-gray-50', 'transition-colors');
    });
  });

  describe('Responsive Design', () => {
    test('applies overflow-x-auto for table responsiveness', () => {
      const props = {
        ...defaultProps,
        documents: [mockDocument]
      };
      
      render(<DocumentSubmissionLog {...props} />);
      
      const tableContainer = screen.getByRole('table').parentElement;
      expect(tableContainer).toHaveClass('overflow-x-auto');
    });
  });

  describe('Accessibility', () => {
    test('has proper button titles for screen readers', () => {
      const props = {
        ...defaultProps,
        documents: [mockDocument]
      };
      
      render(<DocumentSubmissionLog {...props} />);
      
      expect(screen.getByTitle('Download')).toBeInTheDocument();
      expect(screen.getByTitle('Edit')).toBeInTheDocument();
      expect(screen.getByTitle('Delete')).toBeInTheDocument();
    });

    test('has proper table structure', () => {
      const props = {
        ...defaultProps,
        documents: [mockDocument]
      };
      
      render(<DocumentSubmissionLog {...props} />);
      
      expect(screen.getByRole('table')).toBeInTheDocument();
      expect(screen.getAllByRole('columnheader')).toHaveLength(5);
      expect(screen.getAllByRole('row')).toHaveLength(2); // header + 1 data row
    });
  });

  describe('Dynamic Content Updates', () => {
    test('updates description based on document count', () => {
      const { rerender } = render(<DocumentSubmissionLog {...defaultProps} />);
      
      expect(screen.getByText('No documents submitted yet')).toBeInTheDocument();
      
      rerender(<DocumentSubmissionLog {...defaultProps} documents={[mockDocument]} />);
      
      expect(screen.getByText('Track all document submissions with upload details')).toBeInTheDocument();
    });
  });

  describe('Component Integration', () => {
    test('renders without crashing with minimal props', () => {
      expect(() => render(<DocumentSubmissionLog />)).not.toThrow();
    });

    test('handles multiple documents correctly', () => {
      const props = {
        ...defaultProps,
        documents: [mockDocument, mockAdminDocument]
      };
      
      render(<DocumentSubmissionLog {...props} />);
      
      expect(screen.getByText('test-document.pdf')).toBeInTheDocument();
      expect(screen.getByText('admin-document.pdf')).toBeInTheDocument();
      expect(screen.getAllByRole('row')).toHaveLength(3); // header + 2 data rows
    });
  });

  describe('Snapshot Testing', () => {
    test('matches snapshot with no documents', () => {
      const { container } = render(<DocumentSubmissionLog {...defaultProps} />);
      expect(container.firstChild).toMatchSnapshot();
    });

    test('matches snapshot with documents', () => {
      const props = {
        ...defaultProps,
        documents: [mockDocument, mockAdminDocument]
      };
      
      const { container } = render(<DocumentSubmissionLog {...props} />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});