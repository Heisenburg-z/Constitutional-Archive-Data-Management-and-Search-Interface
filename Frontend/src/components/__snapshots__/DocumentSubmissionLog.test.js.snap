// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentSubmissionLog Component Snapshot Testing matches snapshot with documents 1`] = `
<div
  class="bg-white rounded-lg shadow-md p-6 border border-gray-100"
>
  <div
    class="flex items-center gap-3 mb-6"
  >
    <div
      class="p-2 rounded-lg bg-blue-50 text-blue-600"
    >
      <div
        data-testid="history-icon"
      />
    </div>
    <div>
      <h2
        class="text-xl font-bold text-gray-800"
      >
        Document Submission Log
      </h2>
      <p
        class="text-sm text-gray-500"
      >
        Track all document submissions with upload details
      </p>
    </div>
  </div>
  <div
    class="overflow-x-auto"
  >
    <table
      class="min-w-full divide-y divide-gray-200"
    >
      <thead
        class="bg-gray-50"
      >
        <tr>
          <th
            class="px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider"
          >
            <div
              class="flex items-center gap-1"
            >
              <div
                data-testid="filetext-icon"
              />
              Document
            </div>
          </th>
          <th
            class="px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider"
          >
            <div
              class="flex items-center gap-1"
            >
              <div
                data-testid="user-icon"
              />
              Submitted By
            </div>
          </th>
          <th
            class="px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider"
          >
            <div
              class="flex items-center gap-1"
            >
              <div
                data-testid="calendar-icon"
              />
              Submission Date
            </div>
          </th>
          <th
            class="px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider"
          >
            Type
          </th>
          <th
            class="px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider"
          >
            Actions
          </th>
        </tr>
      </thead>
      <tbody
        class="bg-white divide-y divide-gray-200"
      >
        <tr
          class="hover:bg-gray-50 transition-colors"
        >
          <td
            class="px-6 py-4 whitespace-nowrap"
          >
            <div
              class="text-sm font-medium text-gray-900"
            >
              admin-document.pdf
            </div>
          </td>
          <td
            class="px-6 py-4 whitespace-nowrap"
          >
            <div
              class="flex items-center"
            >
              <div
                class="text-sm"
              >
                <div
                  class="font-medium text-gray-900"
                >
                  Admin
                   
                  User
                </div>
                <div
                  class="text-xs text-gray-500"
                >
                  admin@example.com
                </div>
                <span
                  class="mt-1 inline-flex items-center px-2 py-0.5 rounded text-xs font-medium bg-green-100 text-green-800"
                >
                  Admin
                </span>
              </div>
            </div>
          </td>
          <td
            class="px-6 py-4 whitespace-nowrap"
          >
            <div
              class="text-sm text-gray-900"
            >
              2024/01/16
            </div>
            <div
              class="text-xs text-gray-500"
            >
              16:45:00
            </div>
          </td>
          <td
            class="px-6 py-4 whitespace-nowrap"
          >
            <span
              class="px-2 py-1 text-xs font-medium rounded-full bg-purple-100 text-purple-800"
            >
              directory
            </span>
          </td>
          <td
            class="px-6 py-4 whitespace-nowrap text-sm font-medium flex gap-2"
          >
            <button
              class="text-blue-600 hover:text-blue-900 p-1.5 rounded hover:bg-blue-50 transition-colors"
              title="Download"
            >
              <div
                data-testid="download-icon"
              />
            </button>
            <button
              class="text-green-600 hover:text-green-900 p-1.5 rounded hover:bg-green-50 transition-colors"
              title="Edit"
            >
              <div
                data-testid="edit-icon"
              />
            </button>
            <button
              class="text-red-600 hover:text-red-900 p-1.5 rounded hover:bg-red-50 transition-colors"
              title="Delete"
            >
              <div
                data-testid="trash-icon"
              />
            </button>
          </td>
        </tr>
        <tr
          class="hover:bg-gray-50 transition-colors"
        >
          <td
            class="px-6 py-4 whitespace-nowrap"
          >
            <div
              class="text-sm font-medium text-gray-900"
            >
              test-document.pdf
              <div
                class="text-xs text-gray-500 mt-1"
              >
                Test Document Title
              </div>
            </div>
          </td>
          <td
            class="px-6 py-4 whitespace-nowrap"
          >
            <div
              class="flex items-center"
            >
              <div
                class="text-sm"
              >
                <div
                  class="font-medium text-gray-900"
                >
                  John
                   
                  Doe
                </div>
                <div
                  class="text-xs text-gray-500"
                >
                  john.doe@example.com
                </div>
              </div>
            </div>
          </td>
          <td
            class="px-6 py-4 whitespace-nowrap"
          >
            <div
              class="text-sm text-gray-900"
            >
              2024/01/15
            </div>
            <div
              class="text-xs text-gray-500"
            >
              12:30:00
            </div>
          </td>
          <td
            class="px-6 py-4 whitespace-nowrap"
          >
            <span
              class="px-2 py-1 text-xs font-medium rounded-full bg-blue-100 text-blue-800"
            >
              file
            </span>
          </td>
          <td
            class="px-6 py-4 whitespace-nowrap text-sm font-medium flex gap-2"
          >
            <button
              class="text-blue-600 hover:text-blue-900 p-1.5 rounded hover:bg-blue-50 transition-colors"
              title="Download"
            >
              <div
                data-testid="download-icon"
              />
            </button>
            <button
              class="text-green-600 hover:text-green-900 p-1.5 rounded hover:bg-green-50 transition-colors"
              title="Edit"
            >
              <div
                data-testid="edit-icon"
              />
            </button>
            <button
              class="text-red-600 hover:text-red-900 p-1.5 rounded hover:bg-red-50 transition-colors"
              title="Delete"
            >
              <div
                data-testid="trash-icon"
              />
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`DocumentSubmissionLog Component Snapshot Testing matches snapshot with no documents 1`] = `
<div
  class="bg-white rounded-lg shadow-md p-6 border border-gray-100"
>
  <div
    class="flex items-center gap-3 mb-6"
  >
    <div
      class="p-2 rounded-lg bg-blue-50 text-blue-600"
    >
      <div
        data-testid="history-icon"
      />
    </div>
    <div>
      <h2
        class="text-xl font-bold text-gray-800"
      >
        Document Submission Log
      </h2>
      <p
        class="text-sm text-gray-500"
      >
        No documents submitted yet
      </p>
    </div>
  </div>
  <div
    class="text-center py-8 text-gray-500"
  >
    <div
      data-testid="filetext-icon"
    />
    <p>
      No documents have been submitted yet
    </p>
  </div>
</div>
`;
